## Erschließungsstrategien für unbekannten Wortschatz von DaF-Lernenden an spanischen Universitäten (2017-19)
# Datenanalyse - Variablen

##### Werkzeuge und Daten importieren

In [1]:
from funktionen import *
import pandas as pd
import matplotlib
import numpy as np
import scipy as sp
import math
from scipy import stats
from sklearn import tree
import seaborn as sns
sns.set(style="whitegrid") #"white", "darkgrid"...
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
Daten1 = pd.read_pickle('D:\Daten\DISS\Daten\Estrategias_Estudiantes_2017-19_ber')

In [3]:
Daten1.shape

(401, 377)

In [4]:
#Daten.head(3)

In [5]:
gloss = pd.read_pickle('D:\Daten\DISS\Daten\Gloss2018')

Nur vollständige Befragungen sollen berücksichtigt werden:

In [6]:
Daten = Daten1[Daten1['Finished']==1].reset_index(drop=True)

Liste erstellen:

In [7]:
Liste_A_Q2 = [col for col in Daten.columns if 'A-Q2' in col]

In [8]:
Liste_A_Q3 = [col for col in Daten.columns if 'A-Q3' in col]
Liste_A_Q3.pop(3)

'A-Q3_3_TEXT'

In [9]:
Liste_C_Q1 = [col for col in Daten.columns if 'C-Q1' in col]

#### Unabhängige Variablen

Studienrichtung:

In [10]:
Daten['D-Q2a_neu'] = 0

j=0
for i in Daten['D-Q2a']:
    if i == 1:
        Daten.loc[j,'D-Q2a_neu'] = 1
    elif i == 2:
        Daten.loc[j,'D-Q2a_neu'] = 2
    elif i == 3 or i == 4:
        Daten.loc[j,'D-Q2a_neu'] = 3
    else:
        Daten.loc[j,'D-Q2a_neu'] = 4
    j = j+1

In [11]:
Daten.rename(columns={'D-Q2a_neu':'V_Studienrichtung'}, inplace=True)

??? Studienjahr (+ Studium):

In [12]:
Daten['D-Q2b_neu'] = 0

j=0
for i in Daten['D-Q2b']:
    if i == 1:
        Daten.loc[j,'D-Q2b_neu'] = 1
    elif i == 2:
        Daten.loc[j,'D-Q2b_neu'] = 2
    elif i == 3:
        Daten.loc[j,'D-Q2b_neu'] = 3
    elif i == 4:
        Daten.loc[j,'D-Q2b_neu'] = 4
    j = j+1

In [13]:
j=0
for i in Daten['D-Q2']:
    if i == 2:
        Daten.loc[j,'D-Q2b_neu'] = 5
    elif i == 3:
        Daten.loc[j,'D-Q2b_neu'] = 5
    j = j+1

In [14]:
Daten.rename(columns={'D-Q2b_neu':'V_Studienjahr+Studium'}, inplace=True)

Niveau des aktuellen Deutschkurses:

In [15]:
Daten['D-Q4_neu'] = 0

j=0
for i in Daten['D-Q4']:
    if i == 1:
        Daten.loc[j,'D-Q4_neu'] = 1
    elif i == 2:
        Daten.loc[j,'D-Q4_neu'] = 2
    elif i == 3:
        Daten.loc[j,'D-Q4_neu'] = 3
    elif i == 4:
        Daten.loc[j,'D-Q4_neu'] = 4
    else:
        Daten.loc[j,'D-Q4_neu'] = 5
    j = j+1

In [16]:
Daten.rename(columns={'D-Q4_neu':'V_Niveau_DK'}, inplace=True)

Alter:

In [17]:
Daten['D-Q5_neu'] = 0

j=0
for i in Daten['D-Q5']:
    if i <= 18:
        Daten.loc[j,'D-Q5_neu'] = 1
    elif i == 19:
        Daten.loc[j,'D-Q5_neu'] = 2
    elif i == 20:
        Daten.loc[j,'D-Q5_neu'] = 3
    else:
        Daten.loc[j,'D-Q5_neu'] = 4
    j = j+1

In [18]:
Daten.rename(columns={'D-Q5_neu':'V_Alter'}, inplace=True)

Geschlecht:

In [19]:
Daten['D-Q6_neu'] = 0

j=0
for i in Daten['D-Q6']:
    if i == 1:
        Daten.loc[j,'D-Q6_neu'] = 1
    elif i == 2:
        Daten.loc[j,'D-Q6_neu'] = 2
    j = j+1

In [20]:
Daten['D-Q6_neu'].value_counts()

1    287
2     91
0     12
Name: D-Q6_neu, dtype: int64

In [21]:
Daten.rename(columns={'D-Q6_neu':'V_Geschlecht'}, inplace=True)

In [22]:
Daten['D-Q6'].value_counts()

1    287
2     91
3      6
0      6
Name: D-Q6, dtype: int64

In [23]:
#Daten.rename(columns={'D-Q6':'V_Geschlecht1'}, inplace=True)

Erstsprache Spanisch:

In [24]:
Daten['D-Q7_neu'] = 0

j=0
for i in Daten['D-Q7']:
    if i == 1:
        Daten.loc[j,'D-Q7_neu'] = 1
    else:
        Daten.loc[j,'D-Q7_neu'] = 2
    j = j+1

j=0
for i in Daten['D-Q7c']:
    if i == 1:
        Daten.loc[j,'D-Q7_neu'] = 1
    j = j+1

j=0
for i in Daten['D-Q7e']:
    if i == 1:
        Daten.loc[j,'D-Q7_neu'] = 1
    j = j+1

363 Studierende haben Spanisch als eine ihrer Erstsprachen angegeben. Die übrigen 25 verteilen sich auf verschiedene Sprachen (13 TN haben diese Frage nicht beantwortet), darunter haben 5 Personen Katalanisch/Valencianisch und 4 Galicisch angegeben.

In [25]:
Daten['D-Q7_neu'].value_counts()

1    363
2     27
Name: D-Q7_neu, dtype: int64

In [26]:
Daten.rename(columns={'D-Q7_neu':'V_Erstsprache_Span'}, inplace=True)

L1-Bi/Trilingualismus:

In [27]:
Daten['D-Q7b'].value_counts()

2    287
1    103
Name: D-Q7b, dtype: int64

103 Personen sind laut eigenen Angaben bi- oder trilingual.

In [28]:
Daten.rename(columns={'D-Q7b':'V_L1-Bi/Trilingualismus'}, inplace=True)

Weitere Sprachkenntnisse:

In [29]:
Spr_cols = ['D-Q7','D-Q7c','D-Q7e','D-Q8_1','D-Q8_2','D-Q8_3','D-Q8_4','D-Q8_5','D-Q8_6','D-Q8_7']

Anzahl genannter Fremdsprachen:

In [30]:
Anzahl_Sprachen = []

for i in Daten.index:
    Spr_arr = Daten[Spr_cols[3:]].loc[i].unique()
    Spr_arr = Spr_arr[Spr_arr != 0]
    Anzahl_Sprachen.append(len(Spr_arr))

In [31]:
Daten['V_Spr_Zahl'] = Anzahl_Sprachen

In [32]:
Daten['V_Spr_Zahl'].describe()

count    390.000000
mean       4.251282
std        1.433509
min        0.000000
25%        3.000000
50%        4.000000
75%        5.000000
max        7.000000
Name: V_Spr_Zahl, dtype: float64

In [33]:
j=0
for i in Daten['V_Spr_Zahl']:
    if i < 3:
        Daten.loc[j,'V_Spr_Zahl'] = 1
    elif i >= 3 and i < 4:
        Daten.loc[j,'V_Spr_Zahl'] = 3
    elif i >= 4 and i < 5:
        Daten.loc[j,'V_Spr_Zahl'] = 4
    elif i >= 5:
        Daten.loc[j,'V_Spr_Zahl'] = 5
    j = j+1

In [34]:
Daten['V_Spr_Zahl'].value_counts()

5    156
4    115
3     85
1     34
Name: V_Spr_Zahl, dtype: int64

Anzahl genannter Sprachen (inkl. Erstsprachen):

In [35]:
Anzahl_Sprachen_ges = []

for i in Daten.index:
    Spr_arr = Daten[Spr_cols].loc[i].unique()
    Spr_arr = Spr_arr[Spr_arr != 0]
    Anzahl_Sprachen_ges.append(len(Spr_arr))

In [36]:
Daten['V_Spr_Zahl_ges'] = Anzahl_Sprachen_ges

In [37]:
Daten['V_Spr_Zahl_ges'].describe()

count    390.000000
mean       5.633333
std        1.688131
min        1.000000
25%        4.000000
50%        5.500000
75%        7.000000
max       10.000000
Name: V_Spr_Zahl_ges, dtype: float64

In [38]:
j=0
for i in Daten['V_Spr_Zahl_ges']:
    if i < 4:
        Daten.loc[j,'V_Spr_Zahl_ges'] = 1
    elif i >= 4 and i < 5:
        Daten.loc[j,'V_Spr_Zahl_ges'] = 4
    elif i >= 5 and i < 7:
        Daten.loc[j,'V_Spr_Zahl_ges'] = 5
    elif i >= 7:
        Daten.loc[j,'V_Spr_Zahl_ges'] = 7
    j = j+1

In [39]:
Daten['V_Spr_Zahl_ges'].value_counts()

5    184
7    107
4     66
1     33
Name: V_Spr_Zahl_ges, dtype: int64

Einzelne Sprachen und Sprachgruppen:

In [40]:
l_germ = [9,15,39,19,36]
l_rom = [13,16,17]
l_slaw = [23,24,25,26,27,28,41,18,44]
l_asiat = [6,7,8,14,22,31,42,45]

In [41]:
Daten['D-Q8_fr'] = 0
Daten['D-Q8_gr'] = 0
Daten['D-Q8_lat'] = 0
Daten['D-Q8_germ'] = 0
Daten['D-Q8_rom'] = 0
Daten['D-Q8_slaw'] = 0
Daten['D-Q8_asiat'] = 0

In [42]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el == 10:
            Daten.loc[i,'D-Q8_fr'] = 1
            pass

In [43]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el == 20:
            Daten.loc[i,'D-Q8_gr'] = 1
            pass

In [44]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el == 21:
            Daten.loc[i,'D-Q8_lat'] = 1
            pass

In [45]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el in l_germ:
            Daten.loc[i,'D-Q8_germ'] = 1
            pass

In [46]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el in l_rom:
            Daten.loc[i,'D-Q8_rom'] = 1
            pass

In [47]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el in l_slaw:
            Daten.loc[i,'D-Q8_slaw'] = 1
            pass

In [48]:
for i in Daten.index:
    for el in Daten.loc[i,Spr_cols]:
        if el in l_asiat:
            Daten.loc[i,'D-Q8_asiat'] = 1
            pass

In [49]:
Daten.rename(columns={'D-Q8_fr':'V_Spr_Fran'}, inplace=True)
Daten.rename(columns={'D-Q8_gr':'V_Spr_Griech'}, inplace=True)
Daten.rename(columns={'D-Q8_lat':'V_Spr_Latein'}, inplace=True)
Daten.rename(columns={'D-Q8_germ':'V_Spr_Germ'}, inplace=True)
Daten.rename(columns={'D-Q8_rom':'V_Spr_Rom'}, inplace=True)
Daten.rename(columns={'D-Q8_slaw':'V_Spr_Slaw'}, inplace=True)
Daten.rename(columns={'D-Q8_asiat':'V_Spr_Asiat'}, inplace=True)

Strategienerwerbskontexte:

In [50]:
erwerb = {'E1': ['B-Q2-1_1','B-Q2-1_2','B-Q2-1_3','B-Q2-1_4','B-Q2-1_5','B-Q2-1_6','B-Q2-1_7'], 
         'E2': ['B-Q2-2_1','B-Q2-2_2','B-Q2-2_3','B-Q2-2_4','B-Q2-2_5','B-Q2-2_6','B-Q2-2_7'],
         'E3': ['B-Q2-3_1','B-Q2-3_2','B-Q2-3_3','B-Q2-3_4','B-Q2-3_5','B-Q2-3_6','B-Q2-3_7'],
         'E4': ['B-Q2-4_1','B-Q2-4_2','B-Q2-4_3','B-Q2-4_4','B-Q2-4_5','B-Q2-4_6','B-Q2-4_7'],
         'E5': ['B-Q2-5_1','B-Q2-5_2','B-Q2-5_3','B-Q2-5_4','B-Q2-5_5','B-Q2-5_6','B-Q2-5_7'],
         'E6': ['B-Q2-6_1','B-Q2-6_2','B-Q2-6_3','B-Q2-6_4','B-Q2-6_5','B-Q2-6_6','B-Q2-6_7'],
         'E7': ['B-Q2-7_1','B-Q2-7_2','B-Q2-7_3','B-Q2-7_4','B-Q2-7_5','B-Q2-7_6','B-Q2-7_7'],
         'E8': ['B-Q2-8_1','B-Q2-8_2','B-Q2-8_3','B-Q2-8_4','B-Q2-8_5'],
         'E9': ['B-Q2-9_1','B-Q2-9_2','B-Q2-9_3','B-Q2-9_4','B-Q2-9_5']}

* 1- En las clases de Lengua y Literatura en el colegio o instituto
* 2- En las clases de inglés en el colegio o instituto
* 3- En otras clases de idiomas en el colegio o instituto
* 4- En clases de alemán anteriores en la Universidad
* 5- En otras asignaturas de mi carrera universitaria
* 6- En clases de lenguas en escuelas de idiomas o academias
* 7- En mi curso de alemán actual
* 8- En mi familia, con mis amigos/as o compañeros/as de clase
* 9- Por mi cuenta, sin ayuda de otras personas
* 10- Otros:
* 11- No sé cómo he aprendido estas estrategias.

In [51]:
Daten['B-Q2_E1'] = 0
Daten['B-Q2_E2'] = 0
Daten['B-Q2_E3'] = 0
Daten['B-Q2_E4'] = 0
Daten['B-Q2_E5'] = 0
Daten['B-Q2_E6'] = 0
Daten['B-Q2_E7'] = 0
Daten['B-Q2_E8'] = 0
Daten['B-Q2_E9'] = 0

In [52]:
add_columns(Daten,erwerb['E1'],'B-Q2_E1')

In [53]:
add_columns(Daten,erwerb['E2'],'B-Q2_E2')

In [54]:
add_columns(Daten,erwerb['E3'],'B-Q2_E3')

In [55]:
add_columns(Daten,erwerb['E4'],'B-Q2_E4')

In [56]:
add_columns(Daten,erwerb['E5'],'B-Q2_E5')

In [57]:
add_columns(Daten,erwerb['E6'],'B-Q2_E6')

In [58]:
add_columns(Daten,erwerb['E7'],'B-Q2_E7')

In [59]:
add_columns(Daten,erwerb['E8'],'B-Q2_E8')

In [60]:
add_columns(Daten,erwerb['E9'],'B-Q2_E9')

In [61]:
Daten.rename(columns={'B-Q2_E1':'V_E_Schule_LL'}, inplace=True)
Daten.rename(columns={'B-Q2_E2':'V_E_Schule_Eng'}, inplace=True)
Daten.rename(columns={'B-Q2_E3':'V_E_Schule_Spr'}, inplace=True)
Daten.rename(columns={'B-Q2_E4':'V_E_Univ_DK'}, inplace=True)
Daten.rename(columns={'B-Q2_E5':'V_E_Univ_andere'}, inplace=True)
Daten.rename(columns={'B-Q2_E6':'V_E_Sprachschule'}, inplace=True)
Daten.rename(columns={'B-Q2_E7':'V_E_DK_aktuell'}, inplace=True)
Daten.rename(columns={'B-Q2_E8':'V_E_Sozial'}, inplace=True)
Daten.rename(columns={'B-Q2_E9':'V_E_autonom'}, inplace=True)

Methoden des Strategienerwerbs:

In [62]:
erwerb1 = {'U1': ['B-Q2-1_1','B-Q2-2_1','B-Q2-3_1','B-Q2-4_1','B-Q2-5_1','B-Q2-6_1','B-Q2-7_1'], 
         'U2': ['B-Q2-1_2','B-Q2-2_2','B-Q2-3_2','B-Q2-4_2','B-Q2-5_2','B-Q2-6_2','B-Q2-7_2'],
         'U3': ['B-Q2-1_3','B-Q2-2_3','B-Q2-3_3','B-Q2-4_3','B-Q2-5_3','B-Q2-6_3','B-Q2-7_3'],
         'U4': ['B-Q2-1_4','B-Q2-2_4','B-Q2-3_4','B-Q2-4_4','B-Q2-5_4','B-Q2-6_4','B-Q2-7_4'],
         'U5': ['B-Q2-1_5','B-Q2-2_5','B-Q2-3_5','B-Q2-4_5','B-Q2-5_5','B-Q2-6_5','B-Q2-7_5'],
         'U6': ['B-Q2-1_6','B-Q2-2_6','B-Q2-3_6','B-Q2-4_6','B-Q2-5_6','B-Q2-6_6','B-Q2-7_6']}

* (1)	Las estrategias se presentaron (y explicaron) en clase.
* (2)	El/La profesor/a dio consejos sobre cómo familiarizarse de manera autónoma con estas estrategias, p. ej. mediante páginas web, libros u otros materiales que explican estrategias de aprendizaje. 
* (3)	Estas estrategias se explicaron y practicaron con ejemplos concretos, realizando tareas, p. ej. leyendo un texto difícil.
* (4)	Se probaron y se compararon diferentes estrategias.
* (5)	Se reflexionó sobre el uso de estas estrategias, p. ej. sobre qué estrategia sería adecuada en una situación y/o para una persona determinada.
* (6)	Durante las clases he aprendido estas estrategias de manera inconsciente, p. ej. mientras leía textos difíciles, sin explicación o indicación del/de la profesor/a.
* (7)	Otros:

In [63]:
Daten['B-Q2_U1'] = 0
Daten['B-Q2_U2'] = 0
Daten['B-Q2_U3'] = 0
Daten['B-Q2_U4'] = 0
Daten['B-Q2_U5'] = 0
Daten['B-Q2_U6'] = 0

In [64]:
add_columns(Daten,erwerb1['U1'],'B-Q2_U1')

In [65]:
add_columns(Daten,erwerb1['U2'],'B-Q2_U2')

In [66]:
add_columns(Daten,erwerb1['U3'],'B-Q2_U3')

In [67]:
add_columns(Daten,erwerb1['U4'],'B-Q2_U4')

In [68]:
add_columns(Daten,erwerb1['U5'],'B-Q2_U5')

In [69]:
add_columns(Daten,erwerb1['U6'],'B-Q2_U6')

In [70]:
Daten.rename(columns={'B-Q2_U1':'V_Met_Erkl'}, inplace=True)
Daten.rename(columns={'B-Q2_U2':'V_Met_Tipps'}, inplace=True)
Daten.rename(columns={'B-Q2_U3':'V_Met_Training'}, inplace=True)
Daten.rename(columns={'B-Q2_U4':'V_Met_Vgl'}, inplace=True)
Daten.rename(columns={'B-Q2_U5':'V_Met_Reflexion'}, inplace=True)
Daten.rename(columns={'B-Q2_U6':'V_Met_unbewusst'}, inplace=True)

#### Mögliche weitere Einflussfaktoren und wichtige Werte für die Analyse

In [71]:
#Daten['Finished'].value_counts()

In [72]:
#Daten.rename(columns={'Finished':'W_Finished'}, inplace=True)

Pilotstudie:

In [73]:
Daten['pilot'].value_counts()

0    324
1     66
Name: pilot, dtype: int64

In [74]:
Daten.rename(columns={'pilot':'W_Pilot'}, inplace=True)

Erhebungssituation:

In [75]:
Daten['D-Q9_1'].value_counts()

1    228
0    162
Name: D-Q9_1, dtype: int64

In [76]:
Daten.rename(columns={'D-Q9_1':'W_Erhebung_im_U'}, inplace=True)
Daten.rename(columns={'D-Q9_2':'W_Erhebung_nicht_U'}, inplace=True)

Version des Fragebogens:

In [77]:
Daten['D-Q9_pap'] = 0

j=0
for i in Daten['IPAddress']:
    if i == '2.154.100.40':
        Daten.loc[j,'D-Q9_pap'] = 1
    j = j+1

In [78]:
Daten[Daten['D-Q9_pap']==1]['D-Q1'].value_counts()

67    54
50     7
20     3
63     1
Name: D-Q1, dtype: int64

In [79]:
Daten.rename(columns={'D-Q9_pap':'W_Papierversion_FB'}, inplace=True)

Universitäten:

In [80]:
Daten['D-Q1_neu'] = 0

j=0
for i in Daten['D-Q1']:
    if i == 63:
        Daten.loc[j,'D-Q1_neu'] = 1
    elif i == 67:
        Daten.loc[j,'D-Q1_neu'] = 2
    elif i == 73:
        Daten.loc[j,'D-Q1_neu'] = 3
    elif i == 29:
        Daten.loc[j,'D-Q1_neu'] = 4
    elif i == 66:
        Daten.loc[j,'D-Q1_neu'] = 5
    elif i == 20:
        Daten.loc[j,'D-Q1_neu'] = 6
    else:
        Daten.loc[j,'D-Q1_neu'] = 7
    j = j+1

In [81]:
Daten.rename(columns={'D-Q1_neu':'W_Univ'}, inplace=True)

In [82]:
Daten['D-Q1_Sal'] = 0
Daten['D-Q1_Sev'] = 0
Daten['D-Q1_Vigo'] = 0
Daten['D-Q1_Gran'] = 0
Daten['D-Q1_Sant'] = 0
Daten['D-Q1_Madrid'] = 0

In [83]:
j=0
for i in Daten['D-Q1']:
    if i == 63:
        Daten.loc[j,'D-Q1_Sal'] = 1
    j = j+1

In [84]:
j=0
for i in Daten['D-Q1']:
    if i == 67:
        Daten.loc[j,'D-Q1_Sev'] = 1
    j = j+1

In [85]:
j=0
for i in Daten['D-Q1']:
    if i == 73:
        Daten.loc[j,'D-Q1_Vigo'] = 1
    j = j+1

In [86]:
j=0
for i in Daten['D-Q1']:
    if i == 29:
        Daten.loc[j,'D-Q1_Gran'] = 1
    j = j+1

In [87]:
j=0
for i in Daten['D-Q1']:
    if i == 66:
        Daten.loc[j,'D-Q1_Sant'] = 1
    j = j+1

In [88]:
j=0
for i in Daten['D-Q1']:
    if i == 20:
        Daten.loc[j,'D-Q1_Madrid'] = 1
    j = j+1

In [89]:
Daten['D-Q1_Madrid'].value_counts()

0    364
1     26
Name: D-Q1_Madrid, dtype: int64

##### Werte für Analyse anpassen

In [90]:
cols = ['V_Spr_Fran','V_Spr_Griech','V_Spr_Latein','V_Spr_Germ','V_Spr_Rom','V_Spr_Slaw','V_Spr_Asiat',
        'V_E_Schule_LL','V_E_Schule_Eng','V_E_Schule_Spr','V_E_Univ_DK','V_E_Univ_andere','V_E_Sprachschule','V_E_DK_aktuell','V_E_Sozial','V_E_autonom',
        'V_Met_Erkl','V_Met_Tipps','V_Met_Training','V_Met_Vgl','V_Met_Reflexion','V_Met_unbewusst',
        'W_Erhebung_im_U', 'W_Erhebung_nicht_U', 'W_Pilot', 'W_Papierversion_FB']

In [91]:
Daten[cols] = Daten[cols].replace(0,2)

#### Abhängige Variablen

In [92]:
strat = {'S1': ['A-Q3_5','A-Q3_6','A-Q3_7','A-Q3_8','A-Q3_9'], 
         'S2': ['A-Q3_1','A-Q3_2','A-Q3_3','A-Q3_4','A-Q3_5'],
         'S3': ['A-Q3_9','A-Q3_10','A-Q3_11','A-Q3_12'],
         'S4': ['A-Q3_13','A-Q3_14','A-Q3_15','A-Q3_16','A-Q3_17','A-Q3_18','A-Q3_19','A-Q3_20'],
         #'S4a': ['A-Q3_13','A-Q3_14','A-Q3_15','A-Q3_16'],
         #'S4b': ['A-Q3_17','A-Q3_18','A-Q3_19','A-Q3_20'],
         'S5': ['A-Q3_21','A-Q3_22','A-Q3_23','A-Q3_24','A-Q3_25'],
         'S6': ['A-Q3_26','A-Q3_27','A-Q3_28','A-Q3_29','A-Q3_30','A-Q3_31']}

* S1: Auf Wortebene - intralingual
* S2: Auf Wortebene - interlingual
* S3: Auf Satz- und Textebene - engerer Kontext
* S4: Auf Satz- und Textebene - weiterer Kontext (S4a: intratextuell, S4b: Bezug zu Fach- und Weltwissen)
* S5: Außersprachlicher Kontext
* S6: Metakognitive Strategien

Ergebnisse der Faktorenanalyse (in R durchgeführt):

In [93]:
fakt = {'F1': ['A-Q3_6','A-Q3_7','A-Q3_8','A-Q3_9','A-Q3_14','A-Q3_15','A-Q3_10'], 
        'F2': ['A-Q3_1','A-Q3_2','A-Q3_3','A-Q3_4','A-Q3_5'],
        'F3': ['A-Q3_10','A-Q3_11','A-Q3_12','A-Q3_13'],
        'F4': ['A-Q3_16','A-Q3_17','A-Q3_18','A-Q3_19','A-Q3_20'],
        'F5': ['A-Q3_21','A-Q3_22','A-Q3_23','A-Q3_24'],
        'F6': ['A-Q3_26','A-Q3_27','A-Q3_28','A-Q3_29','A-Q3_30','A-Q3_31']}

* F1(MR2): intralinguale Hinweise (Wortbildung, morphosyntaktische Regularitäten, Orthographie) + Konnektoren, Erklärungshilfen
* F2(MR3): interlinguale Hinweise (inkl. Ethymologie)
* F3(MR4): engerer Kontext, Analyse auf Satzebene (Syntax, semantische Relationen)
* F4(MR1): Kontextwissen (Textsorte, behandeltes Thema, kulturelles Wissen...)
* F5(MR6): Bilder, Grafiken, Symbole, Zahlen, Kommunikationssituation
* F6(MR5): metakognitive Strategien

Anpassung der Strategiengruppen nach Erkenntnissen der Faktorenanalyse?

problematisch: Item 14, 15; 10; 13, 16; 25?

##### Scores

Score für alle Likert-Items in A_Q3:

In [94]:
Score = []
for i in Daten.index:
    Score.append(sum(Daten[Liste_A_Q3].loc[i]))

In [95]:
Daten['A-Q4_Score'] = Score

Scores für die einzelnen Strategiengruppen in A_Q3:

In [96]:
Score_S1 = []
for i in Daten.index:
    Score_S1.append(sum(Daten[strat['S1']].loc[i]))

Daten['A-Q4_Score_S1'] = Score_S1

In [97]:
Score_S2 = []
for i in Daten.index:
    Score_S2.append(sum(Daten[strat['S2']].loc[i]))

Daten['A-Q4_Score_S2'] = Score_S2

In [98]:
Score_S3 = []
for i in Daten.index:
    Score_S3.append(sum(Daten[strat['S3']].loc[i]))

Daten['A-Q4_Score_S3'] = Score_S3

In [99]:
Score_S4 = []
for i in Daten.index:
    Score_S4.append(sum(Daten[strat['S4']].loc[i]))

Daten['A-Q4_Score_S4'] = Score_S4

In [100]:
Score_S5 = []
for i in Daten.index:
    Score_S5.append(sum(Daten[strat['S5']].loc[i]))

Daten['A-Q4_Score_S5'] = Score_S5

In [101]:
Score_S6 = []
for i in Daten.index:
    Score_S6.append(sum(Daten[strat['S6']].loc[i]))

Daten['A-Q4_Score_S6'] = Score_S6

##### Mindestanzahl an Strategien

Mindestanzahl an Strategien nach Strategiengruppen:

In [102]:
Daten['A-Q4_S1'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S1']]
    if el[el>=3].count() >= 2:
        Daten.loc[i,'A-Q4_S1'] = 1

In [103]:
Daten['A-Q4_S2'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S2']]
    if el[el>=3].count() >= 2:
        Daten.loc[i,'A-Q4_S2'] = 1

In [104]:
Daten['A-Q4_S3'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S3']]
    if el[el>=3].count() >= 2:
        Daten.loc[i,'A-Q4_S3'] = 1

In [105]:
Daten['A-Q4_S4'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S4']]
    if el[el>=3].count() >= 4:
        Daten.loc[i,'A-Q4_S4'] = 1

In [106]:
Daten['A-Q4_S5'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S5']]
    if el[el>=3].count() >= 2:
        Daten.loc[i,'A-Q4_S5'] = 1

In [107]:
Daten['A-Q4_S6'] = 0

for i in Daten.index:
    el = Daten.loc[i,strat['S6']]
    if el[el>=3].count() >= 3:
        Daten.loc[i,'A-Q4_S6'] = 1

In [108]:
strat_gr = ['A-Q4_S1','A-Q4_S2','A-Q4_S3','A-Q4_S4','A-Q4_S5','A-Q4_S6']

In [109]:
Daten[strat_gr].sum()

A-Q4_S1    362
A-Q4_S2    361
A-Q4_S3    333
A-Q4_S4    357
A-Q4_S5    362
A-Q4_S6    373
dtype: int64

In [110]:
Stratgr_sum = []
for i in Daten.index:
    Stratgr_sum.append(sum(Daten[strat_gr].loc[i]))

Daten['A-Q4_Stratgr'] = Stratgr_sum

In [111]:
Daten['A-Q4_Stratgr'].value_counts()

6    256
5     92
4     32
3      6
2      2
1      2
Name: A-Q4_Stratgr, dtype: int64

Kombinierter Strategieneinsatz:

In [112]:
Daten['A-Q4_Stratgr_all'] = 0

j=0
for i in Daten['A-Q4_Stratgr']:
    if i == 6:
        Daten.loc[j,'A-Q4_Stratgr_all'] = 1
    j = j+1

In [113]:
Daten['A-Q4_Stratgr_all'].value_counts()

1    256
0    134
Name: A-Q4_Stratgr_all, dtype: int64

#### Auswahl der Variablen für die Analyse

In [114]:
Liste_A_Q3 = [col for col in Daten.columns if 'A-Q3' in col]

In [115]:
Liste_A_Q4 = [col for col in Daten.columns if 'A-Q4' in col]

In [116]:
Liste_Var = [col for col in Daten.columns if 'V_' in col]

In [117]:
Liste_Var

['V_L1-Bi/Trilingualismus',
 'V_Studienrichtung',
 'V_Studienjahr+Studium',
 'V_Niveau_DK',
 'V_Alter',
 'V_Geschlecht',
 'V_Erstsprache_Span',
 'V_Spr_Zahl',
 'V_Spr_Zahl_ges',
 'V_Spr_Fran',
 'V_Spr_Griech',
 'V_Spr_Latein',
 'V_Spr_Germ',
 'V_Spr_Rom',
 'V_Spr_Slaw',
 'V_Spr_Asiat',
 'V_E_Schule_LL',
 'V_E_Schule_Eng',
 'V_E_Schule_Spr',
 'V_E_Univ_DK',
 'V_E_Univ_andere',
 'V_E_Sprachschule',
 'V_E_DK_aktuell',
 'V_E_Sozial',
 'V_E_autonom',
 'V_Met_Erkl',
 'V_Met_Tipps',
 'V_Met_Training',
 'V_Met_Vgl',
 'V_Met_Reflexion',
 'V_Met_unbewusst']

In [118]:
def df_spearman(data,lst):

    var1 = 'V_Alter'
    var2 = 'V_Studienjahr+Studium'
    var3 = 'V_Niveau_DK'
    var4 = 'V_Spr_Zahl'
    var5 = 'V_Spr_Zahl_ges'

    Alter = []
    for col in lst:
        corr1, p1 = stats.spearmanr(data[col],data[var1])
        Alter.append(corr1)
   
    Studienjahr = []
    for col in lst:
        corr2, p2 = stats.spearmanr(data[col],data[var2])
        Studienjahr.append(corr2)

    Niveau_Deutschkurs = []
    for col in lst:    
        corr3, p3 = stats.spearmanr(data[col],data[var3])
        Niveau_Deutschkurs.append(corr3)
    
    Spr_Zahl = []
    for col in lst:    
        corr4, p4 = stats.spearmanr(data[col],data[var4])
        Spr_Zahl.append(corr4)
    
    Spr_Zahl_ges = []
    for col in lst:    
        corr5, p5 = stats.spearmanr(data[col],data[var5])
        Spr_Zahl_ges.append(corr5)
    
    df = pd.DataFrame(Alter)
    df.rename(columns={0:'Alter'}, inplace=True)
    df['Studienjahr'] = Studienjahr
    df['Niveau_Deutschkurs'] = Niveau_Deutschkurs
    df['Spr_Zahl'] = Spr_Zahl
    df['Spr_Zahl_ges'] = Spr_Zahl_ges

    df.index = [lst[ind] for ind in df.index]
    df = df.style.applymap(lambda x: 'background-color : salmon' if abs(x) > 0.5 
                           else 'background-color: lightsalmon' if abs(x) > 0.3 and abs(x) <= 0.5
                           else 'background-color: peachpuff' if abs(x) > 0.1 and abs(x) <= 0.3
                           else 'background-color : white')
    
    return df

In [119]:
df_spearman(Daten,('V_Studienrichtung','V_Studienjahr+Studium','V_Niveau_DK','V_Alter','V_Spr_Zahl','V_Spr_Zahl_ges'))

In [120]:
Liste_Var.pop(2)

'V_Studienjahr+Studium'

In [121]:
Liste_Var.pop(7)

'V_Spr_Zahl_ges'

In [122]:
Liste_Var2 = [col for col in Daten.columns if 'W_' in col]

In [123]:
Liste_Var2

['W_Erhebung_im_U',
 'W_Erhebung_nicht_U',
 'W_Pilot',
 'W_Papierversion_FB',
 'W_Univ']

In [124]:
Daten_Analyse = Daten[Liste_A_Q2 + Liste_A_Q3 + Liste_A_Q4 + Liste_C_Q1 + Liste_Var + Liste_Var2]

In [125]:
Daten_Analyse.shape

(390, 95)

In [126]:
Daten_Analyse.to_pickle('D:\Daten\DISS\Daten\Estrategias_Estudiantes_Analyse')

In [127]:
Daten_Analyse.to_excel('D:\Daten\DISS\Daten\Estrategias_Estudiantes_Analyse.xlsx',encoding='utf-16')

In [128]:
Daten_Analyse.to_csv('D:\Daten\DISS\Daten\Estrategias_Estudiantes_Analyse.csv',encoding='utf-16')